www/js/diary/list/DateSelect.tsx

```diff
@@ -4,88 +4,86 @@
     when we have fully migrated to React Native.
 */
 
-import React, { useEffect, useState } from "react";
-import { angularize } from "../../angular-react-helper";
-import { func, object } from "prop-types";
+import React, { useEffect, useState, useMemo, useContext } from "react";
+import { View, Text } from "react-native";
 import moment from "moment";
+import color from "color";
+import { LabelTabContext } from "../LabelTab";
+import { DatePickerModal } from "react-native-paper-dates";
+import { Divider, Button, IconButton, useTheme } from "react-native-paper";
+import i18next from "i18next";
+import { useTranslation } from "react-i18next";
 
 const DateSelect = ({ tsRange, loadSpecificWeekFn }) => {
 
-  const [dateRange, setDateRange] = useState('-\n-');
-  const [selDate, setSelDate] = useState('-');
+  const { pipelineRange } = useContext(LabelTabContext);
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const [open, setOpen] = React.useState(false);
+  const [dateRange, setDateRange] = useState(['-', '-']);
+  const [selDate, setSelDate] = useState(null);
+  const minMaxDates = useMemo(() => ({
+    startDate: new Date(pipelineRange?.start_ts * 1000),
+    endDate: new Date(pipelineRange?.end_ts * 1000),
+  }), [pipelineRange]);
 
   useEffect(() => {
     if (!tsRange.oldestTs) return;
-    const start = moment.unix(tsRange.oldestTs).format('YYYY-MM-DD');
-    const end = moment.unix(tsRange.latestTs).format('YYYY-MM-DD');
-    setDateRange(start + '\n' + end);
-    const mid = moment.unix((tsRange.oldestTs + tsRange.latestTs) / 2).format('YYYY-MM-DD');
-    setSelDate(mid);
+    const displayStartTs = Math.max(tsRange.oldestTs, pipelineRange.start_ts);
+    const displayStartDate = moment.unix(displayStartTs).format('L');
+
+    let displayEndDate;
+    if (tsRange.latestTs < pipelineRange.end_ts) {
+      displayEndDate = moment.unix(tsRange.latestTs).format('L');
+    } else {
+      displayEndDate = t('diary.today');
+    }
+    setDateRange([displayStartDate, displayEndDate]);
+
+    const mid = (tsRange.oldestTs + tsRange.latestTs) / 2;
+    const d = new Date(Math.min(mid, pipelineRange.end_ts) * 1000);
+    setSelDate(d);
```

Comment:
The latest day you can select in the datepicker is the last processed trip. I don't think you should be able to select a day that hasn't been processed yet. 
Since it queries an entire week, selecting the last processed day also means any draft trips should be included too

Reply:
> The latest day you can select in the datepicker is the last processed trip.

I wasn't asking about the latest but the default `const d = new Date(Math.min(mid, pipelineRange.end_ts) * 1000);`

Reply:
The reason for the `Math.min` is so the initial selection is constrained to the same range. If the midpoint is after pipelineRange, we don't want to use the midpoint.

Most noticable when loading test trips from 2016 - we don't want the datepicker to initialize in 2019 or something